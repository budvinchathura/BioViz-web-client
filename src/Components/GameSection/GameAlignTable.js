import React from 'react';
import {Icon} from '@material-ui/core';
import PropTypes from 'prop-types';
import DnaIcon from '../../assets/icons/dna.svg';
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
    table: {
        maxWidth: 1400,
        overflowX: 'auto',
        display: 'block',
    },
    iconTable: {
        display: 'inline-block',
        float: 'left',
    },
}));

export default function GameAlignTable(props) {
    const classes = useStyles();
    return (
        <div>
            <table className={classes.iconTable}>
                <tbody>
                    <tr>
                        <td style={{minWidth: 50}}>
                            <Icon><img src={DnaIcon} alt="seq 1" /></Icon>
                        </td>
                    </tr>
                    <tr>
                        <td style={{minWidth: 50}}>
                            <Icon><img src={DnaIcon} alt="seq 2" /></Icon>
                        </td>
                    </tr>
                </tbody>
            </table>
            <table className={classes.table}>
                <tbody>
                    <tr>
                        {props.align1}
                    </tr>
                    <tr>
                        {props.align2}
                    </tr>
                    <tr>
                        {props.indexLine}
                    </tr>
                </tbody>
            </table>
        </div>
    );
}

GameAlignTable.propTypes = {
    align1: PropTypes.array,
    align2: PropTypes.array,
    indexLine: PropTypes.array,
};
