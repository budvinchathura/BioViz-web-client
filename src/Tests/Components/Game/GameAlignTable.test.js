import React from 'react';
React.useLayoutEffect = React.useEffect;
import renderer from 'react-test-renderer';
import GameAlignTable from '../../../Components/GameSection/GameAlignTable';

describe('GameAlignTable Component', ()=>{
    const testProp = {
        align1: ['A', 'G', 'T'],
        align2: ['C', 'G', 'A'],
        indexLine: [1, 2, 3],
    };
    it('matches snapshot', ()=>{
        const tree = renderer.create(<GameAlignTable {...testProp} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
