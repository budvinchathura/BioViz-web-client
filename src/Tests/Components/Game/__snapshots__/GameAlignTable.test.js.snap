// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameAlignTable Component matches snapshot 1`] = `
<div>
  <table
    className="makeStyles-iconTable-2"
  >
    <tbody>
      <tr>
        <td
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
          >
            <img
              alt="seq 1"
              src="dna.svg"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
          >
            <img
              alt="seq 2"
              src="dna.svg"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="makeStyles-table-1"
  >
    <tbody>
      <tr>
        A
        G
        T
      </tr>
      <tr>
        C
        G
        A
      </tr>
      <tr>
        1
        2
        3
      </tr>
    </tbody>
  </table>
</div>
`;
